test/test_interpreter.cpp

```diff
@@ -626,6 +642,51 @@ TEST_SUITE("xmagics_apply"){
         manager.apply("%dummy", kernel_res);
         REQUIRE(kernel_res["status"] == "error");
     } 
+
+    TEST_CASE("good_status_line") {
+
+        xcpp::xpreamble_manager preamble_manager;
+
+        preamble_manager.register_preamble("magics", new xcpp::xmagics_manager());
+
+        preamble_manager["magics"].get_cast<xcpp::xmagics_manager>().register_magic("magic2", MyMagicCell());
+
+        nl::json kernel_res;
+
+        preamble_manager["magics"].get_cast<xcpp::xmagics_manager>().apply("%%magic2 qwerty", kernel_res);
+
+        REQUIRE(kernel_res["status"] == "ok");
+    } 
+
+    TEST_CASE("good_status_cell") {
+
+        xcpp::xpreamble_manager preamble_manager;
+
+        preamble_manager.register_preamble("magics", new xcpp::xmagics_manager());
+
+        preamble_manager["magics"].get_cast<xcpp::xmagics_manager>().register_magic("magic1", MyMagicLine());
+
+        nl::json kernel_res;
+
+        preamble_manager["magics"].get_cast<xcpp::xmagics_manager>().apply("%magic1 qwerty", kernel_res);
+
+        REQUIRE(kernel_res["status"] == "ok");
+    } 
+
+    TEST_CASE("cell magic with empty cell body") {
+        xcpp::xmagics_manager manager;
+
+        std::stringstream ss;
+        auto cerr_buff = std::cerr.rdbuf();
+        std::cerr.rdbuf(ss.rdbuf());
```

Comment:
I thought the doctest infrastructure had a way to capture the standard error and output streams but could not find anything in the repository or the documentation. @JohanMabille, do you happen to know?

Reply:
The Doctest testing framework does not have built-in support for capturing standard output or standard error streams. I had to manually redirect this stream and capture them.

Reply:
Ok, that's really unfortunate. Can you extract that logic into a RAII object so that future tests that require such a thing to reuse your code?

Reply:
Will add